test/sbvr/migrations.coffee

```diff
@@ -0,0 +1,110 @@
+fs = require('fs')
+typeVocab = fs.readFileSync(require.resolve('@resin/sbvr-types/Type.sbvr'))
+test = require('./test')(typeVocab)
+
+describe 'migrations', ->
+	test.migration '''
+			Term:      name
+				Concept Type: Short Text (Type)
+			Term:      person
+
+			Fact Type: person has name
+				Necessity: each person has exactly one name
+		''',
+		'''
+			Term:      person
+		''', [ '''
+			ALTER TABLE "person"
+				DROP COLUMN IF EXISTS "name";\n\t
+		''' ]
+
+	test.migration '''
+			Term:      person
+		''', '''
+			Term:      name
+				Concept Type: Short Text (Type)
+			Term:      person
+
+			Fact Type: person has name
+				Necessity: each person has exactly one name
+		''', [ '''
+		ALTER TABLE "person"
+			ADD COLUMN IF NOT EXISTS "name" VARCHAR(255) NOT NULL;\n\t
+		''' ]
+
+	test.migration '''
+			Term:      name
+				Synonym: full name
+				Concept Type: Short Text (Type)
+			Term:      person
+
+			Fact Type: person has name
+				Necessity: each person has exactly one name
+		''', '''
+			Term:      full  name
+				Concept Type: Short Text (Type)
+			Term:      person
+
+			Fact Type: person has full name
+				Necessity: each person has exactly one full name
+		''', [ '''
+		ALTER TABLE "person"
+			RENAME COLUMN "name" TO "full name";\n\t
+		''' ]
+
+	test.migration '''
+			Term:      name
+				Concept Type: Short Text (Type)
+			Term:      person
+			Fact Type: person has name
+				Necessity: each person has exactly one name
+		''',
+		'''
+			Term:      name
+				Concept Type: Short Text (Type)
+			Term:      years of experience
+				Concept Type: Integer (Type)
+			Term:      person
+			Fact Type: person has name
+				Necessity: each person has exactly one name
+			Fact Type: person has years of experience
+				Necessity: each person has exactly one years of experience
+		''', [ '''
+		ALTER TABLE "person"
+			ADD COLUMN IF NOT EXISTS "years of experience" INTEGER NOT NULL;\n\t
+		''' ]
+
+	test.migration '''
+		Term: actor
+		''',
+		'''
+			Term: actor
+			Term: user
+				Concept Type: actor
+		''', [ '''
+		CREATE TABLE IF NOT EXISTS "user" (
+			"created at" TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+		,	"id" SERIAL NOT NULL PRIMARY KEY
+		,	"actor" INTEGER NOT NULL
+		,	FOREIGN KEY ("actor") REFERENCES "actor" ("id")
+		);
+		''' ]
+
+	test.migration '''
+		Term: pilot
+		Term: plane
+
+		Fact type:  pilot can fly plane
+			Synonymous Form: plane can be flown by pilot
+		''',
+		'''
+		Term: pilot
+		Term: plane
+
+		Fact type:  plane can be flown by pilot
+		''', [ '''
+			ALTER TABLE "pilot-can fly-plane"
+				RENAME TO "plane-can be flown by-pilot";
+				RENAME COLUMN "pilot" TO "can be flown by-pilot";
```

Comment:
I'm pretty sure this is invalid and should be
```
ALTER TABLE "plane-can be flown by-pilot"
RENAME COLUMN "pilot" TO "can be flown by-pilot";

Reply:
The first instruction is there to rename the the table representing the relation. I used https://www.postgresql.org/docs/9.1/static/sql-altertable.html for reference but I think that it is pretty consistently supported by different dbs.

Reply:
Yes, that reference should be valid in general, my point is that the generated SQL is invalid according to that reference.  Try running the create schema statements for the first model to set it up and then running the migration statements, I'm pretty sure they will fail

Reply:
Oh, also the change was referring to just the line I commented on, for reference I would expect the full block to be:
```coffee
[ '''
	ALTER TABLE "pilot-can fly-plane"
	RENAME TO "plane-can be flown by-pilot";
	ALTER TABLE "plane-can be flown by-pilot"
	RENAME COLUMN "pilot" TO "can be flown by-pilot";
	ALTER TABLE "plane-can be flown by-pilot"
	RENAME COLUMN "can fly-plane" TO "plane";
''' ]
```
or
```coffee
[ '''
	ALTER TABLE "pilot-can fly-plane"
	RENAME TO "plane-can be flown by-pilot";
''', '''
	ALTER TABLE "plane-can be flown by-pilot"
	RENAME COLUMN "pilot" TO "can be flown by-pilot";
''', '''
	ALTER TABLE "plane-can be flown by-pilot"
	RENAME COLUMN "can fly-plane" TO "plane";
''' ]
```